File: src/com/adventofcode/december00/MainProgram.java
package com.adventofcode.december00;

public class MainProgram {

    public static void main(String[] args) {
        new MainProgram();
    }

    //-----------------------------------------------------------

    private BaseClass baseClass;

    private static final boolean test = false;

    public MainProgram() {
        String fileName = getFileName(test);
        Input input = new Input(fileName);
        baseClass = input.getBaseClass();

        long output = run();
        System.out.println("Puzzle output : " + output);
    }

    public long run() {
        long sum = 0;
        return sum;
    }

    //------------------------------------------------------------------------------------------------------------------

    private String getFileName(boolean test) {
        int day = getDayOfMonthFromClassName();
        return "input" + String.format("%02d", day) + "_" + (test ? "example" : "1");
    }

    private int getDayOfMonthFromClassName() {
        String monthName = "december";
        int i = this.getClass().getName().indexOf(monthName);
        int j = this.getClass().getName().lastIndexOf(".");
        String dayOfMonth = this.getClass().getName().substring(i + monthName.length(), j);
        return Integer.parseInt(dayOfMonth);
    }

}

File: src/com/adventofcode/december00/MainProgram.kt
package com.adventofcode.december00

fun main() {
    MainProgram().MainProgram()
}

class MainProgram {
    private val test = false

    fun MainProgram() {
        val fileName = getFileName(test)
        val input = Input(fileName)
        input.getInputList().forEach { println(it) }
        val baseClass = input.getBaseClass()
        val output = run()
        println("Puzzle output : $output")
    }

    fun run(): Long {
        return 0
    }

    //------------------------------------------------------------------------------------------------------------------

    private fun getFileName(test: Boolean): String {
        val day = getDayOfMonthFromClassName()
        return "input" + String.format("%02d", day) + "_" + if (test) "example" else "1"
    }

    private fun getDayOfMonthFromClassName(): Int {
        val monthName = "december"
        val i = this.javaClass.name.indexOf(monthName)
        val j = this.javaClass.name.lastIndexOf(".")
        val dayOfMonth = this.javaClass.name.substring(i + monthName.length, j)
        return dayOfMonth.toInt()
    }
}